File: replace_text.py
import os
import re

def process_file(file_path):
    # Only process text files like liquid, json, css, js, md
    if not re.search(r'\.(liquid|json|css|js|md)$', file_path):
        return

    with open(file_path, 'r', encoding='utf-8') as f:
        content = f.read()

    original_content = content

    # Replace in specific order to prevent partial replacement bugs
    content = re.sub(r'RAFTER M HAT COMPANY', 'MOON RIDGE COMPANY', content)
    content = re.sub(r'Rafter M Hat Company', 'Moon Ridge Company', content)
    content = re.sub(r'Rafter M Hat Co\.', 'Moon Ridge Co.', content)
    content = re.sub(r'Rafter M Hat', 'Moon Ridge', content)
    content = re.sub(r'RAFTER M', 'MOON RIDGE', content)
    content = re.sub(r'Rafter M', 'Moon Ridge', content)

    if content != original_content:
        with open(file_path, 'w', encoding='utf-8') as f:
            f.write(content)
        print('Updated:', file_path)

for root, _, files in os.walk('.'):
    if '.git' in root:
        continue
    for file in files:
        process_file(os.path.join(root, file))

print("Replacement complete.")

File: replace_text.js
const fs = require('fs');
const path = require('path');

function walkDir(dir, callback) {
  fs.readdirSync(dir).forEach(f => {
    let dirPath = path.join(dir, f);
    let isDirectory = fs.statSync(dirPath).isDirectory();
    if (isDirectory) {
      if (f !== 'node_modules' && f !== '.git') {
        walkDir(dirPath, callback);
      }
    } else {
      callback(path.join(dir, f));
    }
  });
}

function processFile(filePath) {
  // Only process text files like liquid, json, css, js, md
  if (!filePath.match(/\.(liquid|json|css|js|md)$/)) return;

  let content = fs.readFileSync(filePath, 'utf8');
  let originalContent = content;

  // Replace in specific order to prevent partial replacement bugs
  content = content.replace(/MOON RIDGE COMPANY/g, 'MOON RIDGE COMPANY');
  content = content.replace(/Moon Ridge Company/g, 'Moon Ridge Company');
  content = content.replace(/Moon Ridge Co\./g, 'Moon Ridge Co.');
  content = content.replace(/Moon Ridge/g, 'Moon Ridge');
  content = content.replace(/MOON RIDGE/g, 'MOON RIDGE');
  content = content.replace(/Moon Ridge/g, 'Moon Ridge');

  if (content !== originalContent) {
    fs.writeFileSync(filePath, content, 'utf8');
    console.log('Updated:', filePath);
  }
}

walkDir('.', processFile);
console.log('Replacement complete.');
